components/marketplace/buy-fractional-order-form.tsx

```diff
@@ -155,11 +155,16 @@ export const BuyFractionalOrderForm = ({
   const percentageAmount = form.watch("percentageAmount");
   const pricePerPercent = form.watch("pricePerPercent");
 
+  const unitsToBuy =
+    BigInt(getUnitsToBuy(percentageAmount)) > BigInt(0)
+      ? getUnitsToBuy(percentageAmount)
+      : "0";
+
   const totalPrice = formatPrice(
     order.chainId,
     getTotalPriceFromPercentage(
```

Comment:
It might make sense to replace this with a `getTotalPriceFromUnits()` method, as it would be a cleaner solution. Currently we have to catch the "0" edge case.

Reply:
Wouldn't it be better for `getUnitsToBuy()` to return a numeric value? When you want the string, you can convert it at the call site.

Reply:
Resolved in https://github.com/hypercerts-org/hypercerts-app/pull/317/commits/4b0ce41c3d12742ed7c3369feedc645e3bc46c2b.